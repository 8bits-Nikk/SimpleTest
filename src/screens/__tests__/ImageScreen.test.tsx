import {renderWithNavigation} from '../../helper/renderWithNavigation';
import ImageScreen from '../ImageScreen';

describe('ImageScreen Tests', () => {
  it('should render correctly', () => {
    const tree = renderWithNavigation(<ImageScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should have ImageView', () => {
    const tree = renderWithNavigation(<ImageScreen />);
    const img = tree.queryByTestId('Image');
    expect(img).not.toBeNull();
  });
  it('should have Image of url', () => {
    const tree = renderWithNavigation(<ImageScreen />);
    const img = tree.getByTestId('Image');
    expect(img.props.source.uri).toBe(
      'https://reactnative.dev/docs/assets/p_cat2.png',
    );
  });
});
