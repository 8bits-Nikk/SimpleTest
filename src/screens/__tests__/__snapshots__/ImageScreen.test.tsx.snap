// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageScreen Tests should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://reactnative.dev/docs/assets/p_cat2.png",
        }
      }
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
      testID="Image"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ddd",
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Previous_btn"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Previous
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="Next_btn"
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          disabled={false}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
